stats/opentelemetry/metricsregistry_test.go

```diff
@@ -0,0 +1,311 @@
+						Value:      1.2,
+					},
+				},
+				Temporality: metricdata.CumulativeTemporality,
+				IsMonotonic: true,
+			},
+		},
+		{
+			Name:        "int-histo",
+			Description: "histogram of call values from tests",
+			Unit:        "int",
+			Data: metricdata.Histogram[int64]{
+				DataPoints: []metricdata.HistogramDataPoint[int64]{
+					{
+						Attributes:   attribute.NewSet(attribute.String("int histo label key", "int histo label value")), // No optional label, not float.
+						Count:        1,
+						Bounds:       bounds,
+						BucketCounts: []uint64{0, 1, 0, 0},
+						Min:          metricdata.NewExtrema(int64(3)),
+						Max:          metricdata.NewExtrema(int64(3)),
+						Sum:          3,
+					},
+				},
+				Temporality: metricdata.CumulativeTemporality,
+			},
+		},
+		{
+			Name:        "float-histo",
+			Description: "histogram of call values from tests",
+			Unit:        "float",
+			Data: metricdata.Histogram[float64]{
+				DataPoints: []metricdata.HistogramDataPoint[float64]{
+					{
+						Attributes:   attribute.NewSet(attribute.String("float histo label key", "float histo label value"), attribute.String("float histo optional label key", "float histo optional label value")),
+						Count:        1,
+						Bounds:       bounds,
+						BucketCounts: []uint64{0, 1, 0, 0},
+						Min:          metricdata.NewExtrema(float64(4.3)),
+						Max:          metricdata.NewExtrema(float64(4.3)),
+						Sum:          4.3,
+					},
+				},
+				Temporality: metricdata.CumulativeTemporality,
+			},
+		},
+		{
+			Name:        "simple-gauge",
+			Description: "the most recent int emitted by test",
+			Unit:        "int",
+			Data: metricdata.Gauge[int64]{
+				DataPoints: []metricdata.DataPoint[int64]{
+					{
+						Attributes: attribute.NewSet(attribute.String("int gauge label key", "int gauge label value")), // No optional label, not float.
+						Value:      8,
+					},
+				},
+			},
+		},
+	}
+
+	clientReader := otelmetric.NewManualReader()
+	clientProvider := otelmetric.NewMeterProvider(otelmetric.WithReader(clientReader))
+
+	// This configures the defaults alongside int counter 3. All the instruments
+	// registered except int counter 2 and 3 are default, so all measurements
+	// recorded should show up in reader collected metrics except those for int
+	// counter 2.
+	// This also only toggles the float count and float histo optional labels,
+	// so only those should show up in metrics emissions. All the required
+	// labels should show up in metrics emissions.
+	mo := MetricsOptions{
+		Metrics:        DefaultMetrics().Add("int-counter-3"),
+		OptionalLabels: []string{"float counter optional label key", "float histo optional label key"},
+		MeterProvider:  clientProvider,
+	}
+	// Don't have access to clientStatsHandler in DialOption, so just create
+	// one directly.
+	csh := &clientStatsHandler{options: Options{MetricsOptions: mo}}
```

Comment:
If you have a `newClientStatsHandler(MetricsOptions)` function (and server equivalent) then you can remove some duplicated code around constructing the reader/provider -- those can be created inside the loop along with MetricsOptions, then the test cases just do:

```go
for _, constructor := range []func(MetricsOptions)otelmetric.MetricsRecorder{newClientStatsHandler,newServerStatsHandler} {
	t.Run(...)
}
```

Reply:
Great suggestion! I had to scale up the interface for initializeMetrics but otherwise took away code duplication.